freqtrade/optimize/backtesting.py

```diff
@@ -336,37 +327,50 @@ def backtest(self, processed: Dict, stake_amount: float,
 
                 indexes[pair] += 1
 
-                if row.buy == 0 or row.sell == 1:
-                    continue  # skip rows where no buy signal or that would immediately sell off
-
-                if (not position_stacking and pair in lock_pair_until
-                        and row.date <= lock_pair_until[pair]):
-                    # without positionstacking, we can only have one open trade per pair.
-                    continue
-
-                if max_open_trades > 0:
-                    # Check if max_open_trades has already been reached for the given date
-                    if not trade_count_lock.get(row.date, 0) < max_open_trades:
-                        continue
-                    trade_count_lock[row.date] = trade_count_lock.get(row.date, 0) + 1
```

Comment:
This line in the prior backtesting causes misalignment in certain backtests.
trade_count_lock[row.date] is incremented again within `_get_sell_trade_entry()` for all candles the trade is open, including the candle the trade was entered.

This is a miss from a previous improvement.

**Impact:**
The last trade-slot may not be filled if the last 2 slots are filled in the same candle.

Assuming a max_open_trades of 3, and 1 open trade with BTC/USDT when we come to the candle at 10:00.
candle 10:00 buys ETH/USDT, and has a buy signal for XRP/USDT.
the XRP trade (assuming int's last in the pairlist) will not be opened, since trade_count_lock is incremented twice for this candle.
A signal at the next candle would enter this trade again - but this can cause misalignments.

To test alignment, best remove this line (removing this line does not break any test).
max_open_trades will not be violated (tested in `test_backtest_multi_pair()`).